src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -881,5 +881,30 @@ public static CurrentProcessingFileInformation convertFromMapOfObjects(
                             .get(PERSISTED_CURRENT_PROCESSING_FILE_HASH)))
                     .build();
         }
+
+        private String getFileHashFromTopic(Topic topic) {
+            Topics topics = topic.parent;
+            Topic hashTopic = topics.find(PERSISTED_CURRENT_PROCESSING_FILE_HASH);
+
+            if (hashTopic != null) {
+                return Coerce.toString(hashTopic);
+            }
+
+            Topic nameTopic = topics.find(PERSISTED_CURRENT_PROCESSING_FILE_NAME);
+
+            if (nameTopic == null) {
+                return null;
+            }
+
+            Path filePath = Paths.get(Coerce.toString(nameTopic));
+            File file = filePath.toFile();
```

Comment:
this can throw, we must handle that. Even though the topic itself is not null, the topic's value might be null.

`Paths.get` can throw `InvalidFilePathException` which must be handled.

Reply:
Agree, will add the exception.